### PR TITLE
new "quick" image_infos lib

### PR DESCRIPTION
**Pull request by AlicVB**
To display quickly important infos about the current image in darkroom.
Capture d’écran_2019-03-09_17-29-30
The goal is to have important information about the current image displayed without the need to expand the metadata view module and search in all the information.

This is a rework of an old try/idea resurrected by some comment here : https://github.com/darktable-org/darktable/issues/2041#issuecomment-459952932

The module can be positioned in 3 places (alterable in prefs) : 
- on top of the navigation
- on top of the histogram
- center on the top line (between filters combobox and the G etc icons
- center on the bottom line (between styles icons and clipping icons)

template (alterable in prefs) : 
- it use the same template system as used in the export lib
- I've extended the predefined variables used there, so this lib as well as export can now use : $(EXIF_EXPOSURE) $(EXIF_APERTURE) $(EXIF_FOCAL_LENGTH) $(EXIF_FOCUS_DISTANCE) $(LONGITUDE) $(LATITUDE) $(ELEVATION)
- Special chars which can be displayed (ex : ° ...) are only printed for the image info lib (not export jobs, to avoid problems)
- It support pango markup
- it support multilines by using the $(NL) variable

Also it would be good to have an option to set place to "none", but I've to find a nice way to not load libs (for now, you can set an empty template and place bottom or top center)

### CODE REVIEW

**Comment by TurboGit** [comment]
Sounds good and so we can certainly remove the image information from the histogram as discussed in #2037.

**Comment by TurboGit** [comment]
> it support multilines by using the $(NL) variable

This seems equivalent to entering directly \n, right?

File: src/libs/tools/image_infos.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2011 Henrik Andersson.
```
Comment (TurboGit) [inline src/libs/tools/image_infos.c]: Have you changed name? :)

**Comment by AlicVB** [comment]
> > it support multilines by using the $(NL) variable
> 
> This seems equivalent to entering directly \n, right?

For some reason, \n doesn't work if entered "manually". It's removed at some point in the variable replacement process, and as I fear to break something for the export, I've preferred to do it like that.

**Comment by TurboGit** [comment]
Ok, fine with me anyway. That was more a question to understand.

**Comment by aurelienpierre** [comment]
> Also it would be good to have an option to set place to "none", but I've to find a nice way to not load libs (for now, you can set an empty template and place bottom or top center)

Just set it wherever you want and use `gtk_widget_set_visible(GTK_WIDGET(widget), FALSE)`

**Comment by TurboGit** [comment]
Yep, agreed with Aurélien, you just have to hide it.

**Comment by Nilvus** [comment]
This PR with #2037 will make the new darktable interface really awesome ! Great work.

**Comment by AlicVB** [comment]
ok, finally, I've just set views to none if position is set to "hidden"

**Comment by TurboGit** [comment]
Thanks ! Looks good.

**Pull request merged**